In [1]:
import pandas as pd
import numpy as np

In [2]:
data=pd.read_csv('database.csv')

/Users/mcndrw/opt/anaconda3/lib/python3.9/site-packages/IPython/core/interactiveshell.py:3444: DtypeWarning: Columns (16) have mixed types.Specify dtype option on import or set low_memory=False.
  exec(code_obj, self.user_global_ns, self.user_ns)


In [4]:
data.head()

Record ID Agency Code Agency Name       Agency Type       City   State  \
0          1     AK00101   Anchorage  Municipal Police  Anchorage  Alaska   
1          2     AK00101   Anchorage  Municipal Police  Anchorage  Alaska   
2          3     AK00101   Anchorage  Municipal Police  Anchorage  Alaska   
3          4     AK00101   Anchorage  Municipal Police  Anchorage  Alaska   
4          5     AK00101   Anchorage  Municipal Police  Anchorage  Alaska   

   Year    Month  Incident              Crime Type  ... Victim Ethnicity  \
0  1980  January         1  Murder or Manslaughter  ...          Unknown   
1  1980    March         1  Murder or Manslaughter  ...          Unknown   
2  1980    March         2  Murder or Manslaughter  ...          Unknown   
3  1980    April         1  Murder or Manslaughter  ...          Unknown   
4  1980    April         2  Murder or Manslaughter  ...          Unknown   

  Perpetrator Sex  Perpetrator Age               Perpetrator Race  \
0            Male               15  Native American/Alaska Native   
1            Male               42                          White   
2         Unknown                0                        Unknown   
3            Male               42                          White   
4         Unknown                0                        Unknown   

  Perpetrator Ethnicity  Relationship         Weapon Victim Count  \
0               Unknown  Acquaintance   Blunt Object            0   
1               Unknown  Acquaintance  Strangulation            0   
2               Unknown       Unknown        Unknown            0   
3               Unknown  Acquaintance  Strangulation            0   
4               Unknown       Unknown        Unknown            0   

  Perpetrator Count Record Source  
0                 0           FBI  
1                 0           FBI  
2                 0           FBI  
3                 0           FBI  
4                 1           FBI  

[5 rows x 24 columns]

In [14]:
data.shape

(638454, 24)

In [11]:
# меняем тип данных, а также ставим нули там, где возраст - пустая строка, чтобы показать, что значеня нет (пропуск)
def int_(x):
    try:
        return int(x)
    except:
        return 0
data['Perpetrator Age']=data['Perpetrator Age'].apply(int_)

# Фильтрация 

In [13]:
# сколько несовершеннолетних преступников, то есть младше 18,
# также убираем нули, так как это неизвестные возрасты преступников
data[(data['Perpetrator Age'] < 18) &(data['Perpetrator Age'] != 0)]

Record ID Agency Code          Agency Name       Agency Type  \
0               1     AK00101            Anchorage  Municipal Police   
31             32     AKASP00  Alaska State Police      State Police   
35             36     AKASP00  Alaska State Police      State Police   
60             61     AL00100            Jefferson           Sheriff   
86             87     AL00102           Birmingham  Municipal Police   
...           ...         ...                  ...               ...   
638355     638356     WIMPD00            Milwaukee  Municipal Police   
638360     638361     WIMPD00            Milwaukee  Municipal Police   
638370     638371     WV00500        Brooke County           Sheriff   
638441     638442     WY00300      Campbell County           Sheriff   
638443     638444     WY01101             Cheyenne  Municipal Police   

             City          State  Year     Month  Incident  \
0       Anchorage         Alaska  1980   January         1   
31         Juneau         Alaska  1980      June         1   
35         Juneau         Alaska  1980    August         1   
60      Jefferson        Alabama  1980  November         1   
86      Jefferson        Alabama  1980  February         3   
...           ...            ...   ...       ...       ...   
638355  Milwaukee      Wisconsin  2014  December         1   
638360  Milwaukee      Wisconsin  2014  December         6   
638370     Brooke  West Virginia  2014       May         1   
638441   Campbell        Wyoming  2014  February         1   
638443    Laramie        Wyoming  2014   October         1   

                        Crime Type  ... Victim Ethnicity Perpetrator Sex  \
0           Murder or Manslaughter  ...          Unknown            Male   
31      Manslaughter by Negligence  ...          Unknown            Male   
35          Murder or Manslaughter  ...          Unknown            Male   
60      Manslaughter by Negligence  ...     Not Hispanic            Male   
86          Murder or Manslaughter  ...          Unknown          Female   
...                            ...  ...              ...             ...   
638355      Murder or Manslaughter  ...          Unknown            Male   
638360      Murder or Manslaughter  ...          Unknown            Male   
638370      Murder or Manslaughter  ...          Unknown          Female   
638441      Murder or Manslaughter  ...     Not Hispanic            Male   
638443      Murder or Manslaughter  ...          Unknown            Male   

        Perpetrator Age               Perpetrator Race Perpetrator Ethnicity  \
0                    15  Native American/Alaska Native               Unknown   
31                    7         Asian/Pacific Islander               Unknown   
35                   17  Native American/Alaska Native               Unknown   
60                   15                          White          Not Hispanic   
86                   17                          Black               Unknown   
...                 ...                            ...                   ...   
638355               17                          Black               Unknown   
638360               17                          Black               Unknown   
638370               16                          White               Unknown   
638441               15                          White          Not Hispanic   
638443               16  Native American/Alaska Native               Unknown   

        Relationship        Weapon Victim Count Perpetrator Count  \
0       Acquaintance  Blunt Object            0                 0   
31            Sister         Rifle            0                 0   
35           Unknown         Rifle            0                 0   
60            Friend       Shotgun            0                 0   
86      Acquaintance       Shotgun            0                 0   
...              ...           ...          ...               ...   
638355      Stranger       Handgun       

In [15]:
33967/638454*100

5.320195346884819

5% преступлений (то есть 33967 из 638454) совершили несовершенноление, младше 18 лет. Интересно посмотреть, против кого совершались эти преступления: против несовершеннолетних или более старшего людей?

In [20]:
data[(data['Perpetrator Age'] < 18) &(data['Perpetrator Age'] != 0) & (data['Victim Age'] < 18)]

Record ID Agency Code          Agency Name       Agency Type  \
0               1     AK00101            Anchorage  Municipal Police   
31             32     AKASP00  Alaska State Police      State Police   
60             61     AL00100            Jefferson           Sheriff   
129           130     AL00102           Birmingham  Municipal Police   
150           151     AL00102           Birmingham  Municipal Police   
...           ...         ...                  ...               ...   
638122     638123     WA03105           Marysville  Municipal Police   
638123     638124     WA03105           Marysville  Municipal Police   
638337     638338     WIMPD00            Milwaukee  Municipal Police   
638355     638356     WIMPD00            Milwaukee  Municipal Police   
638360     638361     WIMPD00            Milwaukee  Municipal Police   

             City       State  Year      Month  Incident  \
0       Anchorage      Alaska  1980    January         1   
31         Juneau      Alaska  1980       June         1   
60      Jefferson     Alabama  1980   November         1   
129     Jefferson     Alabama  1980     August         6   
150     Jefferson     Alabama  1980   November         2   
...           ...         ...   ...        ...       ...   
638122  Snohomish  Washington  2014    October         1   
638123  Snohomish  Washington  2014   November         1   
638337  Milwaukee   Wisconsin  2014  September         1   
638355  Milwaukee   Wisconsin  2014   December         1   
638360  Milwaukee   Wisconsin  2014   December         6   

                        Crime Type  ... Victim Ethnicity Perpetrator Sex  \
0           Murder or Manslaughter  ...          Unknown            Male   
31      Manslaughter by Negligence  ...          Unknown            Male   
60      Manslaughter by Negligence  ...     Not Hispanic            Male   
129         Murder or Manslaughter  ...     Not Hispanic            Male   
150         Murder or Manslaughter  ...     Not Hispanic            Male   
...                            ...  ...              ...             ...   
638122      Murder or Manslaughter  ...     Not Hispanic            Male   
638123      Murder or Manslaughter  ...     Not Hispanic            Male   
638337      Murder or Manslaughter  ...          Unknown            Male   
638355      Murder or Manslaughter  ...          Unknown            Male   
638360      Murder or Manslaughter  ...          Unknown            Male   

        Perpetrator Age               Perpetrator Race Perpetrator Ethnicity  \
0                    15  Native American/Alaska Native               Unknown   
31                    7         Asian/Pacific Islander               Unknown   
60                   15                          White          Not Hispanic   
129                  17                          White          Not Hispanic   
150                  16                          Black          Not Hispanic   
...                 ...                            ...                   ...   
638122               15  Native American/Alaska Native          Not Hispanic   
638123               15  Native American/Alaska Native          Not Hispanic   
638337               16                          Black               Unknown   
638355               17                          Black               Unknown   
638360               17                          Black               Unknown   

        Relationship        Weapon Victim Count Perpetrator Count  \
0       Acquaintance  Blunt Object            0                 0   
31            Sister         Rifle            0                 0   
60            Friend       Shotgun            0                 0   
129     Acquaintance       Shotgun            0                 0   
150     Acquaintance       Handgun            0                 0   
...              ...           ...          ...               ...   
638122        Friend       Handgun            2                 0

In [21]:
10730/33967*100

31.589483910854653

32% преступлений из тех, которые были совершены преступниками младше 18, направлены против несовершеннолетних. Получается, что оставшиеся 68% преступлений были направлены на более старших жертв. Видим интересную закономерность: несовершеннолетние чаще всего совершают преступления против старших

In [31]:
data['Perpetrator Sex'].unique()

array(['Male', 'Unknown', 'Female'], dtype=object)

In [26]:
data[data['Perpetrator Sex']=='Male']

Record ID Agency Code      Agency Name       Agency Type       City  \
0               1     AK00101        Anchorage  Municipal Police  Anchorage   
1               2     AK00101        Anchorage  Municipal Police  Anchorage   
3               4     AK00101        Anchorage  Municipal Police  Anchorage   
5               6     AK00101        Anchorage  Municipal Police  Anchorage   
6               7     AK00101        Anchorage  Municipal Police  Anchorage   
...           ...         ...              ...               ...        ...   
638446     638447     WY01300   Natrona County           Sheriff    Natrona   
638447     638448     WY01301           Casper  Municipal Police    Natrona   
638448     638449     WY01301           Casper  Municipal Police    Natrona   
638450     638451     WY01700  Sheridan County           Sheriff   Sheridan   
638452     638453     WY01800  Sublette County           Sheriff   Sublette   

          State  Year      Month  Incident              Crime Type  ...  \
0        Alaska  1980    January         1  Murder or Manslaughter  ...   
1        Alaska  1980      March         1  Murder or Manslaughter  ...   
3        Alaska  1980      April         1  Murder or Manslaughter  ...   
5        Alaska  1980        May         1  Murder or Manslaughter  ...   
6        Alaska  1980        May         2  Murder or Manslaughter  ...   
...         ...   ...        ...       ...                     ...  ...   
638446  Wyoming  2014  September         1  Murder or Manslaughter  ...   
638447  Wyoming  2014       June         1  Murder or Manslaughter  ...   
638448  Wyoming  2014     August         1  Murder or Manslaughter  ...   
638450  Wyoming  2014       June         1  Murder or Manslaughter  ...   
638452  Wyoming  2014   December         1  Murder or Manslaughter  ...   

       Victim Ethnicity Perpetrator Sex  Perpetrator Age  \
0               Unknown            Male               15   
1               Unknown            Male               42   
3               Unknown            Male               42   
5               Unknown            Male               36   
6               Unknown            Male               27   
...                 ...             ...              ...   
638446     Not Hispanic            Male               48   
638447     Not Hispanic            Male               22   
638448          Unknown            Male               67   
638450          Unknown            Male               57   
638452     Not Hispanic            Male               31   

                     Perpetrator Race Perpetrator Ethnicity  Relationship  \
0       Native American/Alaska Native               Unknown  Acquaintance   
1                               White               Unknown  Acquaintance   
3                               White               Unknown  Acquaintance   
5                               White               Unknown  Acquaintance   
6                               Black               Unknown          Wife   
...                               ...                   ...           ...   
638446                          White          Not Hispanic        Mother   
638447                          White               Unknown    Girlfriend   
638448                          Black               Unknown        Friend   
638450                          White               Unknown  Acquaintance   
638452                          White          Not Hispanic      Stranger   

               Weapon Victim Count Perpetrator Count Record Source  
0        Blunt Object            0                 0           FBI  
1       Strangulation            0                 0           FBI  
3       Strangulation            0                 0           FBI  
5               Rifle            0                 0           FBI  
6               Knife            0                 0           FBI  
...               ...          ...               ...           ...  
638446        Handgun        

In [27]:
data[data['Perpetrator Sex']=='Female']

Record ID Agency Code               Agency Name       Agency Type  \
15             16     AK00101                 Anchorage  Municipal Police   
27             28     AKASP00       Alaska State Police      State Police   
28             29     AKASP00       Alaska State Police      State Police   
46             47     AKASP00       Alaska State Police      State Police   
48             49     AL00100                 Jefferson           Sheriff   
...           ...         ...                       ...               ...   
638433     638434     WVWSP53  State Police: Upperglade      State Police   
638434     638435     WVWSP55       State Police: Welch      State Police   
638435     638436     WVWSP55       State Police: Welch      State Police   
638451     638452     WY01701                  Sheridan  Municipal Police   
638453     638454     WY01902              Rock Springs  Municipal Police   

              City          State  Year      Month  Incident  \
15       Anchorage         Alaska  1980     August         3   
27          Juneau         Alaska  1980   February         2   
28          Juneau         Alaska  1980   February         3   
46          Juneau         Alaska  1980   December         2   
48       Jefferson        Alabama  1980      March         1   
...            ...            ...   ...        ...       ...   
638433     Webster  West Virginia  2014      March         1   
638434    McDowell  West Virginia  2014    October         1   
638435    McDowell  West Virginia  2014    October         1   
638451    Sheridan        Wyoming  2014  September         1   
638453  Sweetwater        Wyoming  2014  September         1   

                        Crime Type  ... Victim Ethnicity Perpetrator Sex  \
15          Murder or Manslaughter  ...          Unknown          Female   
27          Murder or Manslaughter  ...          Unknown          Female   
28          Murder or Manslaughter  ...          Unknown          Female   
46      Manslaughter by Negligence  ...          Unknown          Female   
48          Murder or Manslaughter  ...          Unknown          Female   
...                            ...  ...              ...             ...   
638433      Murder or Manslaughter  ...          Unknown          Female   
638434      Murder or Manslaughter  ...          Unknown          Female   
638435      Murder or Manslaughter  ...          Unknown          Female   
638451      Murder or Manslaughter  ...          Unknown          Female   
638453      Murder or Manslaughter  ...     Not Hispanic          Female   

        Perpetrator Age               Perpetrator Race Perpetrator Ethnicity  \
15                   29                          Black               Unknown   
27                   28  Native American/Alaska Native               Unknown   
28                   36  Native American/Alaska Native               Unknown   
46                   27  Native American/Alaska Native               Unknown   
48                   38                          White               Unknown   
...                 ...                            ...                   ...   
638433               30                          White               Unknown   
638434               35                          White               Unknown   
638435               35                          White               Unknown   
638451               22         Asian/Pacific Islander               Unknown   
638453               24                          White          Not Hispanic   

       Relationship        Weapon Victim Count Perpetrator Count Record Source  
15       Ex-Husband       Handgun            0                 0           FBI  
27          Husband       Handgun            0                 0           FBI  
28          Brother         Rifle            0                 0           FBI  
46          Husband         Rifle            0                 0           FBI  
48          Husband       Shotgun  

In [29]:
48548/638454*100

7.603993396548538

In [30]:
399541/638454*100

62.57944973326192

По общим данным всех годов получается, что мужчины совершают преступления почти в 8 раз чаще, чем женщины (женщины совершили 48548 преступлений - 7.6%, а мужчины - 399541 - 62.6% преступлений )

Интересно, а если пол неизвестен, получается, что преступление не раскрыто?

In [33]:
data[(data['Perpetrator Sex']=='Unknown')].shape[0]

190365

In [32]:
data[(data['Perpetrator Sex']=='Unknown')&(data['Crime Solved']=='No')]

Record ID Agency Code               Agency Name       Agency Type  \
2               3     AK00101                 Anchorage  Municipal Police   
4               5     AK00101                 Anchorage  Municipal Police   
8               9     AK00101                 Anchorage  Municipal Police   
10             11     AK00101                 Anchorage  Municipal Police   
14             15     AK00101                 Anchorage  Municipal Police   
...           ...         ...                       ...               ...   
638418     638419     WV04401                   Spencer  Municipal Police   
638436     638437     WVWSP60  State Police: Williamson      State Police   
638437     638438     WVWSP65               Sp: Hundred      State Police   
638438     638439     WVWSP65               Sp: Hundred      State Police   
638449     638450     WY01500               Park County           Sheriff   

             City          State  Year    Month  Incident  \
2       Anchorage         Alaska  1980    March         2   
4       Anchorage         Alaska  1980    April         2   
8       Anchorage         Alaska  1980     June         2   
10      Anchorage         Alaska  1980     July         1   
14      Anchorage         Alaska  1980   August         2   
...           ...            ...   ...      ...       ...   
638418      Roane  West Virginia  2014      May         1   
638436      Mingo  West Virginia  2014     July         1   
638437       Wood  West Virginia  2014  January         1   
638438       Wood  West Virginia  2014  January         2   
638449       Park        Wyoming  2014  January         1   

                    Crime Type  ... Victim Ethnicity Perpetrator Sex  \
2       Murder or Manslaughter  ...          Unknown         Unknown   
4       Murder or Manslaughter  ...          Unknown         Unknown   
8       Murder or Manslaughter  ...          Unknown         Unknown   
10      Murder or Manslaughter  ...          Unknown         Unknown   
14      Murder or Manslaughter  ...          Unknown         Unknown   
...                        ...  ...              ...             ...   
638418  Murder or Manslaughter  ...          Unknown         Unknown   
638436  Murder or Manslaughter  ...          Unknown         Unknown   
638437  Murder or Manslaughter  ...          Unknown         Unknown   
638438  Murder or Manslaughter  ...          Unknown         Unknown   
638449  Murder or Manslaughter  ...         Hispanic         Unknown   

        Perpetrator Age Perpetrator Race Perpetrator Ethnicity  Relationship  \
2                     0          Unknown               Unknown       Unknown   
4                     0          Unknown               Unknown       Unknown   
8                     0          Unknown               Unknown       Unknown   
10                    0          Unknown               Unknown       Unknown   
14                    0          Unknown               Unknown       Unknown   
...                 ...              ...                   ...           ...   
638418                0          Unknown               Unknown       Unknown   
638436                0          Unknown               Unknown       Unknown   
638437               30          Unknown               Unknown  Acquaintance   
638438               50          Unknown               Unknown  Acquaintance   
638449                0          Unknown               Unknown       Unknown   

         Weapon Victim Count Perpetrator Count Record Source  
2       Unknown            0                 0           FBI  
4       Unknown            0                 1           FBI  
8       Firearm            0                 0           FBI  
10      Unknown            0                 1           FBI  
14      Handgun            0                 0           FBI  
...         ...          ...               ...           ...  
638418  Unknown            0                 0           FBI  
638436    Drugs     

In [35]:
190218/190365*100

99.92277992277992

Более 99.9% преступлений, совершенных неизвестным полом, не раскрыты. Мы оказались правы. 

In [34]:
data[(data['Perpetrator Sex']=='Unknown')&(data['Crime Solved']=='Yes')]

Record ID Agency Code      Agency Name       Agency Type        City  \
405788     405789     FL01300       Miami-Dade     County Police        Dade   
405796     405797     FL01300       Miami-Dade     County Police        Dade   
405972     405973     FL01602     Jacksonville  Municipal Police       Duval   
406255     406256     FL04800    Orange County           Sheriff      Orange   
406259     406260     FL04800    Orange County           Sheriff      Orange   
...           ...         ...              ...               ...         ...   
627925     627926     FL05027     Palm Springs  Municipal Police  Palm Beach   
627977     627978     FL05214   St. Petersburg  Municipal Police    Pinellas   
628035     628036     FL05601      Fort Pierce  Municipal Police   St. Lucie   
628040     628041     FL05601      Fort Pierce  Municipal Police   St. Lucie   
628063     628064     FL05800  Sarasota County           Sheriff    Sarasota   

          State  Year      Month  Incident              Crime Type  ...  \
405788  Florida  2000       July        55  Murder or Manslaughter  ...   
405796  Florida  2000     August        63  Murder or Manslaughter  ...   
405972  Florida  2000    January       239  Murder or Manslaughter  ...   
406255  Florida  2000  September       522  Murder or Manslaughter  ...   
406259  Florida  2000    October       526  Murder or Manslaughter  ...   
...         ...   ...        ...       ...                     ...  ...   
627925  Florida  2014    October       978  Murder or Manslaughter  ...   
627977  Florida  2014    January        30  Murder or Manslaughter  ...   
628035  Florida  2014    January        88  Murder or Manslaughter  ...   
628040  Florida  2014      April        93  Murder or Manslaughter  ...   
628063  Florida  2014   December       116  Murder or Manslaughter  ...   

       Victim Ethnicity Perpetrator Sex  Perpetrator Age Perpetrator Race  \
405788          Unknown         Unknown                0          Unknown   
405796          Unknown         Unknown                0          Unknown   
405972          Unknown         Unknown                0          Unknown   
406255          Unknown         Unknown                0          Unknown   
406259          Unknown         Unknown                0          Unknown   
...                 ...             ...              ...              ...   
627925          Unknown         Unknown                0          Unknown   
627977          Unknown         Unknown                0          Unknown   
628035          Unknown         Unknown                0          Unknown   
628040          Unknown         Unknown                0          Unknown   
628063          Unknown         Unknown                0          Unknown   

       Perpetrator Ethnicity Relationship        Weapon Victim Count  \
405788               Unknown      Unknown       Handgun            1   
405796               Unknown      Unknown       Handgun            1   
405972               Unknown      Unknown       Handgun            1   
406255               Unknown      Unknown       Handgun            1   
406259               Unknown      Unknown       Unknown            1   
...                      ...          ...           ...          ...   
627925               Unknown      Unknown  Blunt Object            1   
627977               Unknown      Unknown         Knife            1   
628035               Unknown      Unknown       Handgun            1   
628040               Unknown      Unknown       Handgun            1   
628063               Unknown      Unknown       Handgun            1   

       Perpetrator Count Record Source  
405788                 0          FOIA  
405796                 0          FOIA  
405972                 0          FOIA  
406255                 0          FOIA  
406259                 0          FOIA  
...                  ...           ...  
627925                 0          FOIA  
627977                 0    

Оставшийся 0.01% раскрытых преступлений имеют, видимо, закрытые данные, так как видим, что многие поля имеют статус Unknown 

In [36]:
data['Year'].mode()

0    1993
dtype: int64

Чаще всего преступления совершались в 1993 году

In [37]:
data[data['Year']==1993]

Record ID Agency Code   Agency Name       Agency Type        City  \
266227     266228     AK00101     Anchorage  Municipal Police   Anchorage   
266228     266229     AK00101     Anchorage  Municipal Police   Anchorage   
266229     266230     AK00101     Anchorage  Municipal Police   Anchorage   
266230     266231     AK00101     Anchorage  Municipal Police   Anchorage   
266231     266232     AK00101     Anchorage  Municipal Police   Anchorage   
...           ...         ...           ...               ...         ...   
290557     290558     WY01600        Platte           Sheriff      Platte   
290558     290559     WY01800      Sublette           Sheriff    Sublette   
290559     290560     WY01900    Sweetwater           Sheriff  Sweetwater   
290560     290561     WY01902  Rock Springs  Municipal Police  Sweetwater   
290561     290562     WY02101      Evanston  Municipal Police       Uinta   

          State  Year      Month  Incident                  Crime Type  ...  \
266227   Alaska  1993        May         2      Murder or Manslaughter  ...   
266228   Alaska  1993    January         1      Murder or Manslaughter  ...   
266229   Alaska  1993   February         1      Murder or Manslaughter  ...   
266230   Alaska  1993   February         2  Manslaughter by Negligence  ...   
266231   Alaska  1993      March         1      Murder or Manslaughter  ...   
...         ...   ...        ...       ...                         ...  ...   
290557  Wyoming  1993        May         1      Murder or Manslaughter  ...   
290558  Wyoming  1993  September         1      Murder or Manslaughter  ...   
290559  Wyoming  1993    January         1      Murder or Manslaughter  ...   
290560  Wyoming  1993      March         1      Murder or Manslaughter  ...   
290561  Wyoming  1993     August         1      Murder or Manslaughter  ...   

       Victim Ethnicity Perpetrator Sex  Perpetrator Age  \
266227          Unknown            Male               20   
266228          Unknown         Unknown                0   
266229          Unknown          Female               39   
266230          Unknown          Female               20   
266231          Unknown         Unknown                0   
...                 ...             ...              ...   
290557          Unknown          Female               19   
290558          Unknown         Unknown                0   
290559          Unknown            Male               28   
290560          Unknown          Female               20   
290561          Unknown            Male               21   

                     Perpetrator Race Perpetrator Ethnicity Relationship  \
266227                          Black               Unknown       Sister   
266228                        Unknown               Unknown      Unknown   
266229                          White               Unknown    Boyfriend   
266230  Native American/Alaska Native               Unknown          Son   
266231                        Unknown               Unknown      Unknown   
...                               ...                   ...          ...   
290557                          White               Unknown    Boyfriend   
290558                        Unknown               Unknown      Unknown   
290559                          White               Unknown     Stranger   
290560                          White               Unknown     Daughter   
290561                          White               Unknown      Ex-Wife   

              Weapon Victim Count Perpetrator Count Record Source  
266227       Handgun            1                 0           FBI  
266228       Handgun            0                 0           FBI  
266229         Knife            0                 0           FBI  
266230      Drowning            0                 0           FBI  
266231         Knife            0                 0           FBI  
...              ...          ...               ...           ...  
290557         Rifle    

In [38]:
24335/638454*100

3.811551027951896

24335 преступлений были совершены в самый "популярный" год в наших данных. Это почти 4% от всех данных. 

In [39]:
data[data['Year']==1993]['State'].value_counts()

California              4369
New York                2511
Texas                   2271
Florida                 1329
Illinois                1013
Michigan                 976
Pennsylvania             834
Louisiana                800
North Carolina           796
Georgia                  764
Maryland                 662
Ohio                     631
Missouri                 560
Virginia                 559
Alabama                  477
Tennessee                468
New Jersey               433
District of Columbia     423
South Carolina           392
Indiana                  368
Arizona                  362
Oklahoma                 301
Washington               287
Arkansas                 264
Wisconsin                241
Kentucky                 240
Mississippi              234
Connecticut              222
Colorado                 221
Massachusetts            210
Oregon                   154
Minnesota                140
Nevada                   137
West Virginia            129
New Mexico    

В "популярном" 1993 году преступления совершались в разных штатах, в топ-5 попали: Калифорния, Нью-Йорк, Техас, Флорида и Иллинойс. Меньше в штате Мэн

In [40]:
data[data['Year']==1993]['Perpetrator Sex'].value_counts()

Male       14842
Unknown     7900
Female      1593
Name: Perpetrator Sex, dtype: int64

В "популярном" 1993 году преступления совершали чаще всего мужчины, в 14 раз меньше - женщины

In [41]:
data[(data['Year']==1993) & (data['Victim Sex']=='Female')]

Record ID Agency Code   Agency Name       Agency Type        City  \
266227     266228     AK00101     Anchorage  Municipal Police   Anchorage   
266235     266236     AK00101     Anchorage  Municipal Police   Anchorage   
266237     266238     AK00101     Anchorage  Municipal Police   Anchorage   
266238     266239     AK00101     Anchorage  Municipal Police   Anchorage   
266247     266248     AK00101     Anchorage  Municipal Police   Anchorage   
...           ...         ...           ...               ...         ...   
290553     290554     WY01200       Lincoln           Sheriff     Lincoln   
290555     290556     WY01301        Casper  Municipal Police     Natrona   
290559     290560     WY01900    Sweetwater           Sheriff  Sweetwater   
290560     290561     WY01902  Rock Springs  Municipal Police  Sweetwater   
290561     290562     WY02101      Evanston  Municipal Police       Uinta   

          State  Year     Month  Incident              Crime Type  ...  \
266227   Alaska  1993       May         2  Murder or Manslaughter  ...   
266235   Alaska  1993       May         2  Murder or Manslaughter  ...   
266237   Alaska  1993       May         4  Murder or Manslaughter  ...   
266238   Alaska  1993       May         5  Murder or Manslaughter  ...   
266247   Alaska  1993   October         4  Murder or Manslaughter  ...   
...         ...   ...       ...       ...                     ...  ...   
290553  Wyoming  1993       May         1  Murder or Manslaughter  ...   
290555  Wyoming  1993  November         1  Murder or Manslaughter  ...   
290559  Wyoming  1993   January         1  Murder or Manslaughter  ...   
290560  Wyoming  1993     March         1  Murder or Manslaughter  ...   
290561  Wyoming  1993    August         1  Murder or Manslaughter  ...   

       Victim Ethnicity Perpetrator Sex  Perpetrator Age  \
266227          Unknown            Male               20   
266235          Unknown            Male               20   
266237          Unknown         Unknown                0   
266238          Unknown          Female               22   
266247          Unknown            Male               32   
...                 ...             ...              ...   
290553          Unknown            Male               47   
290555          Unknown            Male               17   
290559          Unknown            Male               28   
290560          Unknown          Female               20   
290561          Unknown            Male               21   

                     Perpetrator Race Perpetrator Ethnicity  Relationship  \
266227                          Black               Unknown        Sister   
266235                          Black               Unknown        Sister   
266237                        Unknown               Unknown       Unknown   
266238  Native American/Alaska Native               Unknown      Stranger   
266247                          Black               Unknown  Acquaintance   
...                               ...                   ...           ...   
290553                          White               Unknown        Family   
290555                          White               Unknown      Stranger   
290559                          White               Unknown      Stranger   
290560                          White               Unknown      Daughter   
290561                          White               Unknown       Ex-Wife   

              Weapon Victim Count Perpetrator Count Record Source  
266227       Handgun            1                 0           FBI  
266235       Handgun            1                 0           FBI  
266237       Handgun            0                 0           FBI  
266238       Unknown            0                 0           FBI  
266247         Knife            0                 0           FBI  
...              ...          ...               ...           ...  
290553       Handgun            0                 0           FBI  
2905

In [42]:
5349/24335*100

21.980686254366137

22%  преступлений в "популярный" 1993 год точно совершены против женщин (есть точная информация). Мы ожидали, что этот процент будет больше

# Создание нового столбца и применение с ним фильтрации

In [59]:
data['Relationship'].unique()

array(['Acquaintance', 'Unknown', 'Wife', 'Stranger', 'Girlfriend',
       'Ex-Husband', 'Brother', 'Stepdaughter', 'Husband', 'Sister',
       'Friend', 'Family', 'Neighbor', 'Father', 'In-Law', 'Son',
       'Ex-Wife', 'Boyfriend', 'Mother', 'Common-Law Husband',
       'Common-Law Wife', 'Stepfather', 'Stepson', 'Stepmother',
       'Daughter', 'Boyfriend/Girlfriend', 'Employer', 'Employee'],
      dtype=object)

In [60]:
family_members = ['Ex-Husband', 'Wife', 'Brother', 'Stepdaughter', 'Husband', 'Sister', 'Family', 'Father', 'In-Law', 'Son','Ex-Wife', 'Mother', 'Common-Law Husband','Common-Law Wife', 'Stepfather', 'Stepson', 'Stepmother','Daughter']
data['Victim_is_Family_member']=data['Relationship'].apply(lambda x: 'Yes' if x in family_members else 'No')

Мы посмотрели, какие типы отношений существуют в таблице, выделили в список family_members тех, кто является членом семьи, далее создали столбец Victim_is_Family_member, где пропишем 'Yes', если жертва является для преступника членом семьи и 'No' в ином случае 

сколько преступлений где преступника больше 1: 

     сколько из них раскрыты (то есть преступник больше 1 и crime soldev равно yes) 
     против кого чаще всего направлено преступление (против семьи или против незнакомцев - по новому столбцу)

# Сортировка

In [53]:
data.sort_values(['Year', 'Incident'], ascending=[True, False])

Record ID Agency Code      Agency Name       Agency Type        City  \
16293       16294     NY03030         New York  Municipal Police    New York   
16292       16293     NY03030         New York  Municipal Police    New York   
16291       16292     NY03030         New York  Municipal Police    New York   
16290       16291     NY03030         New York  Municipal Police    New York   
16289       16290     NY03030         New York  Municipal Police    New York   
...           ...         ...              ...               ...         ...   
638450     638451     WY01700  Sheridan County           Sheriff    Sheridan   
638451     638452     WY01701         Sheridan  Municipal Police    Sheridan   
638452     638453     WY01800  Sublette County           Sheriff    Sublette   
638453     638454     WY01902     Rock Springs  Municipal Police  Sweetwater   
627947     627948     FL05100     Pasco County           Sheriff       Pasco   

           State  Year      Month  Incident                  Crime Type  ...  \
16293   New York  1980   December       239  Manslaughter by Negligence  ...   
16292   New York  1980   December       238      Murder or Manslaughter  ...   
16291   New York  1980   December       237      Murder or Manslaughter  ...   
16290   New York  1980   December       236      Murder or Manslaughter  ...   
16289   New York  1980   December       235      Murder or Manslaughter  ...   
...          ...   ...        ...       ...                         ...  ...   
638450   Wyoming  2014       June         1      Murder or Manslaughter  ...   
638451   Wyoming  2014  September         1      Murder or Manslaughter  ...   
638452   Wyoming  2014   December         1      Murder or Manslaughter  ...   
638453   Wyoming  2014  September         1      Murder or Manslaughter  ...   
627947   Florida  2014  September         0      Murder or Manslaughter  ...   

       Victim Ethnicity Perpetrator Sex  Perpetrator Age  \
16293           Unknown            Male               21   
16292           Unknown         Unknown                0   
16291           Unknown         Unknown                0   
16290           Unknown         Unknown                0   
16289           Unknown         Unknown                0   
...                 ...             ...              ...   
638450          Unknown            Male               57   
638451          Unknown          Female               22   
638452     Not Hispanic            Male               31   
638453     Not Hispanic          Female               24   
627947          Unknown            Male                0   

              Perpetrator Race Perpetrator Ethnicity  Relationship  \
16293                    Black               Unknown        Friend   
16292                  Unknown               Unknown       Unknown   
16291                  Unknown               Unknown       Unknown   
16290                  Unknown               Unknown       Unknown   
16289                  Unknown               Unknown      Stranger   
...                        ...                   ...           ...   
638450                   White               Unknown  Acquaintance   
638451  Asian/Pacific Islander               Unknown      Daughter   
638452                   White          Not Hispanic      Stranger   
638453                   White          Not Hispanic      Daughter   
627947                   White               Unknown       Unknown   

              Weapon Victim Count Perpetrator Count Record Source  
16293        Handgun            0                 0           FBI  
16292        Handgun            0                 0           FBI  
16291        Handgun            0                 0           FBI  
16290        Handgun            0                 0           FBI  
16289        Handgun            0                 0           FBI  
...              ...          ...               ...           ...  
638450       Handgun            0               

К 1980 году в Нью-Йорке было совершено 239 преступлений (самый большой номер инцидента)

In [55]:
data.sort_values('Perpetrator Age', ascending=False)

Record ID Agency Code      Agency Name       Agency Type  \
284238     284239     OH04807           Toledo  Municipal Police   
284248     284249     OH04807           Toledo  Municipal Police   
417054     417055     WV02500           Marion           Sheriff   
634095     634096     OH02500  Franklin County           Sheriff   
421347     421348     FL04102        Bradenton  Municipal Police   
...           ...         ...              ...               ...   
288218     288219     TXDPD00           Dallas  Municipal Police   
288216     288217     TXDPD00           Dallas  Municipal Police   
288215     288216     TXDPD00           Dallas  Municipal Police   
288211     288212     TXDPD00           Dallas  Municipal Police   
319227     319228     DCMPD00       Washington  Municipal Police   

                        City                 State  Year     Month  Incident  \
284238                 Lucas                  Ohio  1993    August         1   
284248                 Lucas                  Ohio  1993   October         4   
417054                Marion         West Virginia  2000  November         1   
634095              Franklin                  Ohio  2014      July         1   
421347               Manatee               Florida  2001       May       425   
...                      ...                   ...   ...       ...       ...   
288218                Dallas                 Texas  1993   January        24   
288216                Dallas                 Texas  1993   January        22   
288215                Dallas                 Texas  1993   January        21   
288211                Dallas                 Texas  1993   January        17   
319227  District of Columbia  District of Columbia  1995   January        14   

                    Crime Type  ... Victim Ethnicity Perpetrator Sex  \
284238  Murder or Manslaughter  ...          Unknown         Unknown   
284248  Murder or Manslaughter  ...          Unknown         Unknown   
417054  Murder or Manslaughter  ...          Unknown            Male   
634095  Murder or Manslaughter  ...          Unknown         Unknown   
421347  Murder or Manslaughter  ...          Unknown         Unknown   
...                        ...  ...              ...             ...   
288218  Murder or Manslaughter  ...     Not Hispanic         Unknown   
288216  Murder or Manslaughter  ...     Not Hispanic         Unknown   
288215  Murder or Manslaughter  ...     Not Hispanic         Unknown   
288211  Murder or Manslaughter  ...         Hispanic         Unknown   
319227  Murder or Manslaughter  ...          Unknown         Unknown   

        Perpetrator Age Perpetrator Race Perpetrator Ethnicity Relationship  \
284238               99          Unknown               Unknown      Unknown   
284248               99          Unknown               Unknown     Stranger   
417054               99            White               Unknown     Stranger   
634095               99          Unknown               Unknown      Unknown   
421347               99          Unknown               Unknown      Unknown   
...                 ...              ...                   ...          ...   
288218                0          Unknown               Unknown      Unknown   
288216                0          Unknown               Unknown      Unknown   
288215                0          Unknown               Unknown      Unknown   
288211                0          Unknown               Unknown      Unknown   
319227                0          Unknown               Unknown      Unknown   

         Weapon Victim Count Perpetrator Count Record Source  
284238  Handgun            0                 1           FBI  
284248  Handgun            0                 0           FBI  
417054    Knife            0                 0           FBI  
634095  Firearm            0                 0           FBI  
421347    Knife            0                 0          FOIA  
...         ...          ...          

При сортировке возраста преступника мы заметили, что есть странные возрасты 99, что также говорит о наличии иных пропусков в возрасте, кроме 0, который мы поставили изначальное

In [56]:
data.sort_values('Victim Count', ascending=False)

Record ID Agency Code    Agency Name       Agency Type         City  \
330101     330102     OK05506  Oklahoma City  Municipal Police     Oklahoma   
330157     330158     OK05506  Oklahoma City  Municipal Police     Oklahoma   
268686     268687     CA01942    Los Angeles  Municipal Police  Los Angeles   
330184     330185     OK05506  Oklahoma City  Municipal Police     Oklahoma   
330185     330186     OK05506  Oklahoma City  Municipal Police     Oklahoma   
...           ...         ...            ...               ...          ...   
220232     220233     AL05904      Alabaster  Municipal Police       Shelby   
220233     220234     AL05904      Alabaster  Municipal Police       Shelby   
220234     220235     AL06000         Sumter           Sheriff       Sumter   
220235     220236     AL06000         Sumter           Sheriff       Sumter   
638453     638454     WY01902   Rock Springs  Municipal Police   Sweetwater   

             State  Year      Month  Incident              Crime Type  ...  \
330101    Oklahoma  1995      April         9  Murder or Manslaughter  ...   
330157    Oklahoma  1995      April        19  Murder or Manslaughter  ...   
268686  California  1993    October         4  Murder or Manslaughter  ...   
330184    Oklahoma  1995      April         6  Murder or Manslaughter  ...   
330185    Oklahoma  1995      April         7  Murder or Manslaughter  ...   
...            ...   ...        ...       ...                     ...  ...   
220232     Alabama  1991    January         1  Murder or Manslaughter  ...   
220233     Alabama  1991       June         1  Murder or Manslaughter  ...   
220234     Alabama  1991      March         1  Murder or Manslaughter  ...   
220235     Alabama  1991   December         1  Murder or Manslaughter  ...   
638453     Wyoming  2014  September         1  Murder or Manslaughter  ...   

       Victim Ethnicity Perpetrator Sex  Perpetrator Age Perpetrator Race  \
330101          Unknown            Male               27            White   
330157          Unknown            Male               27            White   
268686         Hispanic            Male               25            White   
330184          Unknown            Male               27            White   
330185          Unknown            Male               27            White   
...                 ...             ...              ...              ...   
220232          Unknown            Male               30            Black   
220233          Unknown            Male                0            Black   
220234          Unknown         Unknown                0          Unknown   
220235          Unknown            Male               40            Black   
638453     Not Hispanic          Female               24            White   

       Perpetrator Ethnicity     Relationship        Weapon Victim Count  \
330101               Unknown         Stranger    Explosives           10   
330157               Unknown         Stranger    Explosives           10   
268686              Hispanic         Stranger          Fire           10   
330184               Unknown         Stranger    Explosives           10   
330185               Unknown         Stranger    Explosives           10   
...                      ...              ...           ...          ...   
220232               Unknown     Acquaintance       Handgun            0   
220233               Unknown           Family         Knife            0   
220234               Unknown          Unknown       Firearm            0   
220235               Unknown  Common-Law Wife       Handgun            0   
638453          Not Hispanic         Daughter  Blunt Object            0   

       Perpetrator Count Record Source  
330101                 2           FBI  
330157                 2           FBI  
268686                 1           FBI  
330184                 2           FBI  
330185                 2           FBI  
...                  ...           ...

Максимальное количество жертв равняется 10, при этом на них нападал 1 преступник или их было двое, видим также, что жертвы для преступников были незнакомцами. 